benchflow-dsl/src/main/scala/cloud/benchflow/dsl/BenchFlowTestAPI.scala

```diff
@@ -50,11 +59,86 @@ object BenchFlowTestAPI {
 
   }
 
-  def validateTest(benchFlowTest: BenchFlowTest): Boolean = {
+  private val userMissing = "Missing user specification. "
+
+  def validateTest(benchFlowTest: BenchFlowTest, validationErrorMessages: StringBuilder): Boolean = {
+
+    validationErrorMessages.append("Semantic validation failed. ")
+    var isValid = true
 
     // TODO - validate semantic in separate function on the object
 
-    false
+    // check that number of users is defined if exploration goal
```

Comment:
Is there a way to enable and execute just the checks that are needed, given the result of a previous check?

Reply:
You would then need to keep track of the errors and then execute just those checks. Why would you want to do that? Currently it is the user's responsibility to ensure it is fixed.